### PR TITLE
Pass ws.Channel to ocppj callbacks

### PR DESCRIPTION
**Pull request by michaelbeaumont**
For my purposes at least, it's enough to have the `NewChargePointHandler` accept `ws.Channel`s, so I haven't changed all of the request callbacks. WDYT? I'll update `2.0` if it looks good. @lorenzodonini

### CODE REVIEW

**Comment by lorenzodonini** [comment]
I like the idea and changes look ok. I just have two considerations:
1. not convinced about the `ChargePointHandler` naming, simply because in OCPP 2.0 charge points are suddenly called charging stations. I would imagine this naming will be kept in future versions, so it would suggest making this "future-proof" from the start. WDYT?
2. I was trying to keep the `ocppXX` layer as independent from the websocket layer as possible (users don't even see it in normal usage). With these changes the separation becomes less obvious. I would love not to extend the `ws.Channel` interface with too many accessors in the future, because the point is to not access the websocket layer directly, but keep the `ocppXX` layer as simple and transparent as possible.

Otherwise I'm fine with adding this as a breaking change for an upcoming release (once the tests are passing).

**Comment by michaelbeaumont** [comment]
1. Fixed
2. I've added a wrapper interface so that at least the transport level is opaque to `ocpp1.6`

File: ocpp1.6/v16.go [hunk 1]
```diff
@@ -17,6 +19,13 @@ import (
 	"github.com/lorenzodonini/ocpp-go/ws"
 )
 
+type ChargePointConnection interface {
+	ID() string
+	TLSConnectionState() *tls.ConnectionState
+}
```
Comment (lorenzodonini) [inline ocpp1.6/v16.go:25]: To clarify my previous comment: if these function names match the ones on the websocket layer, then `ChargePointConnection` automatically becomes a `ws.Channel`. So on this layer we can work with a high-level interface, which just hides the underlying websocket layer, although it is the same object.

Comment (michaelbeaumont) [reply ocpp1.6/v16.go:25]: I didn't want to break the `ws.Channel` interface, but I'll just fix it there then :grin:

Comment (michaelbeaumont) [reply ocpp1.6/v16.go:25]: simplified it

**Review by lorenzodonini — APPROVED** [review]

**Pull request merged**